docs/troubleshooting/agent.md

```diff
@@ -0,0 +1,79 @@
+# The Switch Agent
+
+## Agent Overview
+
+The agent is a process authored by Hedgehog that creates and enforce the
```

Comment:
Do we explicitly mention who authored the other components? What info does it bring?

Reply:
My goal here was to let people know what is HH value add and what is the underlying OS. They are aware they are looking at our software and not SONiC.